src/indexpackage.jl

```diff
@@ -0,0 +1,73 @@
+module SymbolServer
+
+using Pkg, SHA
+using Base: UUID
+
+current_package_name = Symbol(ARGS[1])
+current_package_version = VersionNumber(ARGS[2])
+current_package_uuid = UUID(ARGS[3])
+current_package_treehash = ARGS[4]
+
+@info "Indexing package $current_package_name $current_package_version..."
+
+# This path will always be mounted in the docker container in which we are running
+store_path = "/symcache"
+
+current_package_versionwithoutplus = replace(string(current_package_version), '+'=>'_')
+cache_package_folder_path = joinpath(store_path, "v1", "packages", string(uppercase(string(current_package_name)[1])), "$(current_package_name)_$current_package_uuid")
+filename_without_extension = "v$(current_package_versionwithoutplus)_$current_package_treehash"
+filename_with_extension = "$filename_without_extension.jstore"
+cache_path = joinpath(cache_package_folder_path, filename_with_extension)
+cache_path_compressed = joinpath(cache_package_folder_path, "$filename_without_extension.tar.gz")
+
+mkpath(cache_package_folder_path)
+
+module LoadingBay end
+
+try
+    Pkg.add(name=string(current_package_name), version=current_package_version)
+catch err
+    @info "Could not install package, exiting"
+    exit(20)
+end
+
+# TODO Make the code below ONLY write a cache file for the package we just added here.
+include("faketypes.jl")
+include("symbols.jl")
+include("utils.jl")
+include("serialize.jl")
+using .CacheStore
+
+# Load package
+m = try
+    LoadingBay.eval(:(import $current_package_name))
+    getfield(LoadingBay, current_package_name)
+catch e
+    @info "Could not load package, exiting."
+    exit(10)
+end
+
+# Get the symbols
+env = getenvtree([current_package_name])
+symbols(env, m)
```

Comment:
when https://github.com/julia-vscode/SymbolServer.jl/pull/203 is merged to master (and the commit pulled across to here) this should be changed to `symbols(env, m, get_return_type = true)`